var/spack/repos/builtin/packages/py-junit2html/package.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyJunit2html(PythonPackage):
+    """
+    Simple self-contained python tool to produce a single html file from a single junit xml file.
+    """
+
+    homepage = "https://gitlab.com/inorton/junit2html"
+
+    pypi = "junit2html/junit2html-31.0.2-py3-none-any.whl"
```

Comment:
As a general rule Spack packages should be built from source.  There are some exceptions, such as proprietary software.

Reply:
You can convert this to grab the source from GitLab.  

You'll need to:
- use the `url` property instead of `pypi`;
- update the version sha256 (`8d90ae83163dde6bf0bde9c3e8d21c0ab0796de7c5f33917cfdbb9d3192122131`);
- add the build dependency (i.e., `py-setuptools`); and
- add the 'build' and 'run' dependency (i.e., `py-jinja2@3:`).

The dependencies are based on the version's `setup.py` file.

There's more information on python packaging at https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html.